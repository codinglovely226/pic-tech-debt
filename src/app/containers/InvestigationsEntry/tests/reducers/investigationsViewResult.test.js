/* eslint-disable */

import investigationsViewResultReducer from '../../reducers/investigationsViewResult';

import {
  hideInvestigationsViewResultModal,
  removeInvestigationsViewResultAttachment,
  resetInvestigationsViewResult,
  showInvestigationsViewResultModal,
  updateInvestigationsViewResultAttachments,
  updateInvestigationsViewResultDescription,
  updateInvestigationsViewResultNotes,
} from '../../actions/case';

describe('investigationsViewResultReducer', () => {
  it('returns the initial state', () => {
    expect(investigationsViewResultReducer(undefined, {})).toMatchSnapshot();
  });
  it('hide investigations view result modal', () => {
    expect(investigationsViewResultReducer({}, hideInvestigationsViewResultModal())).toMatchSnapshot();
  });
  it('remove investigations view result attachment', () => {
    expect(investigationsViewResultReducer({'attachments': [0,1,2,3,4] }, removeInvestigationsViewResultAttachment(1))).toMatchSnapshot();
  });
  updateInvestigationsViewResultAttachments,
  updateInvestigationsViewResultDescription,
  updateInvestigationsViewResultNotes,
  it('reset investigations view result', () => {
    expect(investigationsViewResultReducer({'attachments': [0,1,2,3,4] }, resetInvestigationsViewResult())).toMatchSnapshot();
  });
  it('show investigations view result modal', () => {
    expect(investigationsViewResultReducer({}, showInvestigationsViewResultModal(1))).toMatchSnapshot();
  });
  it('update investigations view result attachments', () => {
    expect(investigationsViewResultReducer({}, updateInvestigationsViewResultAttachments([1, 2, 3]))).toMatchSnapshot();
  });
  it('update investigations view result description', () => {
    expect(investigationsViewResultReducer({}, updateInvestigationsViewResultDescription('Test Description'))).toMatchSnapshot();
  });
  it('update investigations view result notes', () => {
    expect(investigationsViewResultReducer({}, updateInvestigationsViewResultNotes('Test Content'))).toMatchSnapshot();
  });
});
