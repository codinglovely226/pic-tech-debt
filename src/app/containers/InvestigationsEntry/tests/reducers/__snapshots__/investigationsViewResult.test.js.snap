exports[`investigationsViewResultReducer hide investigations view result modal 1`] = `
Object {
  "attachments": Array [],
  "content": "",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer remove investigations view result attachment 1`] = `
Object {
  "attachments": Array [
    0,
    2,
    3,
    4,
  ],
  "content": "",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer reset investigations view result 1`] = `
Object {
  "attachments": Array [],
  "content": "",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer returns the initial state 1`] = `
Object {
  "attachments": Array [],
  "content": "",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer show investigations view result modal 1`] = `
Object {
  "attachments": Array [],
  "content": "",
  "description": "",
  "requestIx": 1,
  "show": true,
}
`;

exports[`investigationsViewResultReducer update investigations view result attachments 1`] = `
Object {
  "attachments": Array [
    1,
    2,
    3,
  ],
  "content": "",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer update investigations view result description 1`] = `
Object {
  "attachments": Array [],
  "content": "",
  "description": "Test Description",
  "requestIx": "",
  "show": false,
}
`;

exports[`investigationsViewResultReducer update investigations view result notes 1`] = `
Object {
  "attachments": Array [],
  "content": "Test Content",
  "description": "",
  "requestIx": "",
  "show": false,
}
`;
